database/migrations/2024_08_01_193038_remove_and_rebuild_wiki_lifecycle_events.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+use App\Jobs\UpdateWikiSiteStatsJob;
+use App\WikiLifecycleEvents;
+use Illuminate\Database\Migrations\Migration;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        WikiLifecycleEvents::all()->map->delete();
```

Comment:
Is `map` expected to be here?

Reply:
It was deliberate but it's perhaps a somewhat confusing shorthand.

You can't simply call `delete()` on a collection but need to all it on each item in the collection. You can read about it here https://laravel.com/docs/10.x/collections#higher-order-messages

Do you think this is too obscure to be understandable and the more simple
```
->map(function (int $item, int $key) {
    $item->delete();
});
```

would be nicer?

Reply:
In the tests I usually use `Something::query()->delete()` which seems to work as intended. Any reason against using this? The map as is is indeed slightly confusing?

Reply:
yeah, that's definitely way better; just hadn't thought of it. It's probably also "more efficient" since it doesn't trigger loading all the Events into memory